manifest: update hal_nordic revision to integrate nrfx 3.12.0

nika-nordic opened the pull request:
New nrfx release contains MDK 8.71.1
and various fixes for nRF devices.

nika-nordic committed:
manifest: update hal_nordic revision to integrate nrfx 3.12.0

New nrfx release contains MDK 8.71.1
and various fixes for nRF devices.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

anangl commented:
Since this version introduces a new driver (nrfx_mramc), I think that a Kconfig option for it and corresponding entries in `modules/hal_nordic/nrfx/CMakeLists.txt` and `modules/hal_nordic/nrfx/nrfx_kconfig.h` should be added.

nika-nordic committed:
drivers: adc: nrf: align to channel-wide burst configuration

Some nRF devices allow to configure burst setting per channel
while others globally.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>

nika-nordic committed:
drivers: adc: nrf: align to non-gain configuration

Some nRF devices do not allow to configure gain.

Signed-off-by: Nikodem Kastelik <nikodem.kastelik@nordicsemi.no>
